### PR TITLE
Adding Ellipse Shape

### PR DESCRIPTION
**Pull request by MicahElizabeth**
Adding the class for Ellipse and a rotate function for shape.

### CODE REVIEW

File: src/app/shapes/Ellipse.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
+import * as Konva from "konva";
+import {Shape} from "./Shape";
+
+export class Ellipse extends Shape{
+
+  public instance: Konva.Ellipse;
```
Comment (MatthewMeacham) [inline src/app/shapes/Ellipse.ts:6]: `instance` is a variable in the Shape superclass, so you shouldn't need to define it here.

File: src/app/shapes/Ellipse.ts [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+  }
+  
+  public getTop(): number {
+    return this.instance.y() - this.instance.radius().y();
+  }
+  
+  private createKonvaEllipse(width: number, height: number): Konva.Ellipse {
+      const strokeWidth = 2;
+    return new Konva.Ellipse({
+        x: (width - (2*strokeWidth) )/ 2 + strokeWidth,
```
Comment (MatthewMeacham) [inline src/app/shapes/Ellipse.ts]: What is happening here? Can we make a private method with a descriptive name to say what is happening here? Also why are we adding the strokeWidth at the end?

File: src/app/shapes/Ellipse.ts (hunk 1 above)
Comment (allisoninman) [inline src/app/shapes/Ellipse.ts:6]: instance is on shape now so remove this here

File: src/app/shapes/Ellipse.ts [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+        y: (height - (2*strokeWidth) ) / 2 + strokeWidth,
+        radius: {
+            x: (width - (2*strokeWidth) ) / 2,
```
Comment (MatthewMeacham) [inline src/app/shapes/Ellipse.ts]: Same thing here, make a private method to describe you're creating the radius here. Similarly, the 2*strokeWidth isn't immediately obvious of why we're doing that. Give a descriptive variable name

File: src/app/shapes/Ellipse.ts (hunk 1 above)
Comment (MicahElizabeth) [reply src/app/shapes/Ellipse.ts:6]: We need to define it so that we can use Ellipse specific functionality.

Comment (MicahElizabeth) [reply src/app/shapes/Ellipse.ts:6]: We need to define it so that we can use Ellipse specific functionality.

**Review by shaminoo — APPROVED** [review]

**Review by allisoninman — APPROVED** [review]

**Pull request merged**